tracker:
  ros__parameters:
    tracker:
      dim_objects: 8
      object_classes: ['void_ignore',
           'bicycle',
           'bus',
           'car',
           'motorcycle',
           'pedestrian',
           'trailer',
           'truck']
      frame_id: 'map'
      asgn_thresh: 10.0
      # trk_timeout: 1.5
      publishers:
        names: ['track_pub']
        track_pub: # Publisher name
          pub_topic: 'tracks'
          msg_type: 'Tracks3D'
          routine: 'PublishTracks'

      # Track management
      trk_mgmt_method: 'count' # count or prob     
      n_birth_min_list: [1, 3, 1, 1, 3, 1, 3, 1]
      n_age_max_list: [2, 2, 2, 2, 2, 2, 2, 2]
      del_thresh_list: [0.] # Placeholder for prob method
      pub_thresh_list: [0.] # Placeholder for prob method
      

    object_properties: # TODO - or motion/process models?
      object_classes: ['void_ignore',
           'bicycle',
           'bus',
           'car',
           'motorcycle',
           'pedestrian',
           'trailer',
           'truck']
      void_ignore:
        model_type: 'ab3dmot'
        proc_var: [.1, .1, .1, .1, .1, .1, .1, .1, 0., 0., 0.]
      bicycle:
        model_type: 'ab3dmot'
        proc_var: [.1, .1, .1, .1, .1, .1, .1, .1, 0., 0., 0.]
      bus:
        model_type: 'ab3dmot'
        proc_var: [.1, .1, .1, .1, .1, .1, .1, .1, 0., 0., 0.]
      car:
        model_type: 'ab3dmot'
        proc_var: [.1, .1, .1, .1, .1, .1, .1, .1, 0., 0., 0.]
      motorcycle:
        model_type: 'ab3dmot'
        proc_var: [.1, .1, .1, .1, .1, .1, .1, .1, 0., 0., 0.]
      pedestrian:
        model_type: 'ab3dmot'
        proc_var: [.1, .1, .1, .1, .1, .1, .1, .1, 0., 0., 0.]
      trailer:
        model_type: 'ab3dmot'
        proc_var: [.1, .1, .1, .1, .1, .1, .1, .1, 0., 0., 0.]
      truck:
        model_type: 'ab3dmot'
        proc_var: [.1, .1, .1, .1, .1, .1, .1, .1, 0., 0., 0.]

    sensors:

      #Declare names of sensors upfront
      sensor_names: ['nuscenes-megvii']
      

      # For each named sensor, give sensor model information
      nuscenes-megvii:
        detector_index: 0
        topic: 'converted_detections'
        msg_type: 'Detections3D'
        dim_classes: 8
        dim_detections: 10
        detection_classes: ['barrier',
              'bicycle',
              'bus',
              'car',
              'construction_vehicle',
              'motorcycle',
              'pedestrian',
              'traffic_cone',
              'trailer',
              'truck']

        # Spatial sensor model
        spatial_sensor_model: [1., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 
                0., 1., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
                0., 0., 1., 0., 0., 0., 0., 0., 0., 0., 0., 
                0., 0., 0., 1., 0., 0., 0., 0., 0., 0., 0.,
                0., 0., 0., 0., 1., 0., 0., 0., 0., 0., 0.,
                0., 0., 0., 0., 0., 1., 0., 0., 0., 0., 0.,
                0., 0., 0., 0., 0., 0., 1., 0., 0., 0., 0.,
                0., 0., 0., 0., 0., 0., 0., 1., 0., 0., 0.]
        spatial_states: 11
        spatial_meas: 8
        sensor_variance: # values from mini_train
          pos: [.195, .195, .0163, # barrier
                .0423,.0424,.0169, #bicycle
                .231, .231, .0878, # bus
                .0747, .0747, .0262, # car
                .0906, .0906, .0491, # construction vehicle
                .0645, .0645, .0208, # motorcycle
                .0341, .0341, .0203, # pedestrian
                .0648, .0648, .00935, # traffic cone
                .0615, .0615, .0230, # trailer
                .115, .115, .0368] # truck
          yaw: [.079, # barrier
                .075, # bicycle
                .0015, # bus
                .21, # car
                .50, # construction_vehicle
                .037, # motorcycle
                .36, # pedestrian
                .39, # traffic cone
                .018, # trailer
                .31] # truck
          size: [.076,.098,.019, # barrier
                 .031, .0098, .018, # bicycle
                 1.1, .043, .061, # bus
                 .076, .017, .018, # car
                 1.4, .18, .17, # construction_vehicle
                 .042, .025, .0068, # motorcycle
                 .025, .011, .011, # pedestrian
                 .019, .093, .012, # traffic cone
                 .11, .071, .038, # trailer
                 .59, .044, .046] # truck
          conf: [.01, # barrier
                .01, # bicycle
                .01, # bus
                .01, # car
                .01, # construction_vehicle
                .01, # motorcycle
                .01, # pedestrian
                .01, # traffic cone
                .01, # trailer
                .01] # truck

        # Semantic sensor model        
        # p(Z_label|X_class): 
        # columns are dets:  barrier / bicycle / bus / car / construction_vehicle / motorcycle / pedestrian / traffic_cone / trailer / truck 
        semantic_sensor_model: [0.57, 0., 0., 0., 0.05, 0., 0., 0.38, 0., 0., # void_ignore - barrier, construction_vehicle, traffic cone
                        0., 1., 0., 0., 0., 0., 0., 0., 0., 0., # bicycle - 20% missed chance, 5% misclassification chance
                        0., 0., 1., 0., 0., 0., 0., 0., 0., 0., # bus
                        0., 0., 0., 1., 0., 0., 0., 0., 0., 0., # car
                        0., 0., 0., 0., 0., 1., 0., 0., 0., 0., # motorcycle
                        0., 0., 0., 0., 0., 0., 1., 0., 0., 0., # pedestrian
                        0., 0., 0., 0., 0., 0., 0., 0., 1., 0., # trailer
                        0.011429, 0.000000, 0.009524, 0.196190, 0.011429, 0.000000, 0.007619, 0.000000, 0.005714, 1.] # truck

        # From mini_train
        p_missed_det_list: [0.19773001205303334, 
                            0.32759344813103736, 
                            0.34198860037998735, 
                            0.2487625618719064, 
                            0.2460127591706539, 
                            0.19629079686726641, 
                            0.41909112767492185, 
                            0.3624554253693327]

        # From mini_train
        hist_bins: [0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 1. ]
        false_pos_hist: [10314.,  1797.,   484.,   177.,    69.,    26.,    11.,     3.,     1., # 'void_ignore',
                        8.27027027, 1.18918919, .364864865, .121621622, .0405405405, .00675675676, .00675675676, 0., 0., # 'bicycle',
                        7.44088896, 1.54900159, 0.51382739, 0.22725374, 0.10694294, 0.06600384, 0.04762303, 0.03592614, 0.01253238, # 'bus',
                        7.64770580, 1.41938577, .497898982, .217716236, .101666714, .0595053442, .0362390366, .0164979272, .00338419019, # 'car',
                        8.00476632, 1.35795931, .398958471, .127543140, .0626682554, .0247142416, .0145637495, .00661988614, .00220662871, # 'motorcycle',
                        7.53462604, 1.72151431, .576546630, .139242844, .0240073869, .00332409972, .000738688827, 0., 0., # 'pedestrian',
                        8.20711809, 1.1805674 , 0.31468531, 0.14213429, 0.06708738, 0.04491444, 0.02302575, 0.0099494 , 0.01051794, # 'trailer',
                        7.76773265, 1.37093510, .456528584, .180587415, .104124500, .0654432600, .0319345118, .0168668196, .00584716413] # 'truck'
        true_pos_hist: [7832., 4657., 4264., 2588., 1419.,  845.,  633.,  914.,  851., # 'void_ignore',
                        3.17926296, 1.67395378, 1.41786384, 1.13678951, 0.93066833, 0.61211743, 0.45596502, 0.29981262, 0.29356652, # 'bicycle',
                        0.84215592, 0.51010587, 0.50529355, 0.53416747, 0.64485082, 0.55341675, 0.74109721, 1.26564004, 4.40327238, # bus
                        1.53139663, 0.86220093, 0.7607068 , 0.73891052, 0.76935876, 0.86486307, 1.14039466, 1.9335463 , 1.39862234, # 'car',
                        2.46430460, 1.17398202, 0.97831835, 0.92014807, 0.64516129, 0.63458488, 0.67689053, 0.867266  , 1.63934426, # 'motorcycle',
                        1.66382902, 2.61909074, 2.94185836, 1.61746060, 0.769063575, 0.328925919, 0.0594095273, 0.000362253215, 0., # 'pedestrian',
                        2.04470199, 1.5397351 , 0.98923841, 0.625     , 0.52566225, 0.60016556, 0.7781457 , 1.09271523, 1.80463576, # 'trailer',
                        2.12145425, 1.15561326, 0.92588893, 0.84798242, 1.00878945, 0.9688374 , 1.01977627, 1.20055933, 0.75109868] # 'truck'